cpp/test/sg/fil_test.cu

```diff
@@ -878,6 +1007,15 @@ std::vector<FilTestParams> predict_dense_inputs = {
                   num_trees   = 3,
                   leaf_algo   = VECTOR_LEAF,
                   num_classes = 4000),
+  FIL_TEST_PARAMS(node_categorical_prob = 0.5, feature_categorical_prob = 0.5),
```

Comment:
Some more tests would definitely be useful.

Reply:
I ran the suite through gcov, and there aren't any new lines, ternary operator branches or enum values that can be further covered with more tests (except the sparse tests which I've added). There may be a couple tests we can add to FIL in general (all invalid parameters), but they don't need categorical features.